synctool/src/main/java/org/duracloud/sync/endpoint/DuraStoreChunkSyncEndpoint.java

```diff
@@ -193,16 +198,81 @@ private ChunksManifest getManifest(String spaceId, String contentId) {
+            Iterator<String> chunkedContentIdIt = store.getSpaceContents(getSpaceId(), contentId + ".dura-");
+            Set<String> chunkedContentIds = new HashSet<>();
+            chunkedContentIdIt.forEachRemaining(id -> chunkedContentIds.add(id));
+            //if there are chunks
+            if (!chunkedContentIds.isEmpty()) {
+                //clean up
+                if (syncFile.length() <= chunkerOptions.getMaxChunkSize()) {
+                    log.info("A chunked version was replaced by an unchunked version of {}/{}",
+                              spaceId, contentId);
+
+                    //if file is less than or equal to max chunk size
+                    //then we can expect the new file is unchunked
+                    //look for any chunked content matching path
+                    //delete all.
+                    chunkedContentIds.stream().forEach(content -> {
+                        deleteContent(spaceId, content, store);
+                    });
+                    log.info("Deleted manifest and all chunks associated with {}/{}",
+                              spaceId, contentId);
+                } else {
+                    log.info("Checking for orphaned chunks associated with {}/{}",
+                             spaceId, contentId);
+
+                    //resolve the set of the chunks in the manifest
+                    ChunksManifest manifest = getManifest(spaceId, contentId);
+                    Set<String> manifestChunks = new HashSet<>();
+                    manifest.getEntries().stream().forEach(entry -> manifestChunks.add(entry.getChunkId()));
+                    //for each chunk in storage, delete if not in the manifest.
+                    chunkedContentIds.stream()
+                                     .filter(chunkedContentId -> !chunkedContentId.endsWith(manifestSuffix))
+                                     .forEach(chunk -> {
+                                         if (!manifestChunks.contains(chunk)) {
+                                             log.debug("Chunk not found in manifest: deleting orphaned chunk ({}/{})",
+                                                       spaceId, chunk);
+                                             deleteContent(spaceId, chunk, store);
+                                         }
+                                     });
+                    //check for an unchunked version and remove it if happens to exist.
+                    if (store.contentExists(spaceId, contentId)) {
+                        deleteContent(spaceId, contentId, store);
+                    }
+                }
+            }
+        } catch (Exception ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+
+    private void deleteContent(String spaceId, String contentId, ContentStore store) {
+        try {
+            store.deleteContent(spaceId, contentId);
+            log.info("Deleted content  ({}/{})", spaceId, contentId);
```

Comment:
Is this log needed if it's being described above? Perhaps this moves to debug level? I just don't want "delete" log messages showing up in the log without some context.

Reply:
Valid point.  I'll change it to debug.